src/test/java/com/google/sps/servlets/DashboardServletTest.java

```diff
@@ -0,0 +1,101 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import com.google.sps.servlets.DBUtilities;
+import org.mockito.Mockito;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.io.StringWriter;
+import java.io.PrintWriter;
+import com.google.appengine.api.datastore.Query;
+import com.google.sps.servlets.Property;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+
+@RunWith(JUnit4.class)
+public final class DashboardServletTest {
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private PrintWriter printWriter  = new PrintWriter(new StringWriter());
+  private HttpServletRequest request;       
+  private HttpServletResponse response; 
+
+  @Before
+  public void setUp() { 
+    request = Mockito.mock(HttpServletRequest.class);
+    response = Mockito.mock(HttpServletResponse.class); 
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testCorrectStatusWhenParameterNull() throws IOException {
+
+    // could also be 'endTime' here
+    Mockito.when(request.getParameter("startTime")).thenReturn(null);
+    Mockito.when(response.getWriter()).thenReturn(printWriter);
+
+    new DashboardServlet().doGet(request, response);
+
+    Mockito.verify(response).setStatus(HttpServletResponse.SC_OK);
+  }
+
+  @Test
+  public void testCorrectStatusWhenParameterEmpty() throws IOException {   
```

Comment:
You probably want to verify in these tests that you are handling empty parameters correctly.  So make sure that the query still goes from 0 if you don't include a start, or to the present if you dont include an end

Reply:
that is handled in the front-end if the date picker is empty and it is sent as a normal query with two parameters